CRM/Utils/Address.php

```diff
@@ -347,10 +347,22 @@ public static function getFormattedBillingAddressFieldsFromParameters($params, $
 
     $addressFields = array();
     foreach ($addressParts as $name => $field) {
-      $addressFields[$name] = CRM_Utils_Array::value($field, $params);
+      $value = CRM_Utils_Array::value($field, $params);
+      $alternateName = 'billing_' . $name . '_id-' . $billingLocationTypeID;
       //Include values which prepend 'billing_' to country and state_province.
-      if (empty($params[$field])) {
-        $addressFields[$name] = CRM_Utils_Array::value('billing_' . $field, $params);
+      if (empty($params[$field]) &&  CRM_Utils_Array::value($alternateName, $params)) {
+        $value = $params[$alternateName];
+      }
+      if (is_numeric($value) && $name == 'state_province' || $name == 'country') {
+        if ($name == 'state_province') {
+          $addressFields[$name] = CRM_Core_PseudoConstant::stateProvince($value);
+        }
+        if ($name == 'country') {
+          $addressFields[$name] = CRM_Core_PseudoConstant::country($value);
```

Comment:
This displays name instead of abbreviations for me too, maybe it should be `stateProvinceAbbreviation()` and `countryIsoCode()` call ?